apps/rush-lib/src/logic/PackageJsonUpdater.ts

```diff
@@ -355,18 +355,28 @@ export class PackageJsonUpdater {
       }
       console.log(`Querying NPM registry for latest version of "${packageName}"...`);
 
+      //if the project exists in the local
+      const rushConfigProjects: RushConfigurationProject[] = this._rushConfiguration.projects;
+      rushConfigProjects.forEach(rushConfigProject => {
+        if(rushConfigProject.packageName === packageName) {
+          selectedVersion = rushConfigProject.packageJson.version;
```

Comment:
You should break out of the loop here.

We also probably want to handle the case where a version is specified on the command-line and it doesn't match the version in the package.json

We also need to handle the case where the dependency that we're adding is marked as a cyclic dependency for the project in `rush.json`.